### PR TITLE
[WIP] Merging public calendar sharing

### PR DESCRIPTION
**Pull request by jbtbnl**
:exclamation: This is a rebase of https://github.com/owncloud/calendar/pull/308.

@MorrisJobke @jancborchardt @georgehrke please check this out. It is my first merge so it might not be perfect :smile: 
- [ ] fix »Share by link« action + icon

### CODE REVIEW

**Comment by jbtbnl** [comment]
@rysiekpl the share-by-link dialog doesn't work well with the new calendar sidebar. [rowfields.php#L51](https://github.com/owncloud/calendar/blob/rysiekpl-master/templates/part.choosecalendar.rowfields.php#L51) 

Any suggestions how to fix this?

**Comment by jancborchardt** [comment]
So this is basically a rebase/merge of https://github.com/owncloud/calendar/pull/308 ? 

AWESOME! :) @rysiekpl is everything how you imagine?

@owncloud/designers @MorrisJobke @DeepDiver1975 @georgehrke we need to review this!

**Comment by georgehrke** [comment]
- [ ] when changing timezone on public link timezone is not set

**Comment by georgehrke** [comment]
- [ ] calendar - owncloud at 10 15 46 pm

File: ajax/event/edit.form.php [hunk 1]
```diff
@@ -1,18 +1,81 @@
...
  * See the COPYING-README file.
  */
 
+// "this is from a link-shared calendar" flag
+$link_shared = false;
 
+// do we have a logged-in user?
+if(OCP\User::isLoggedIn()) {
+	// yeah, we do. check if the app is enabled
+	OCP\JSON::checkAppEnabled('calendar');
```
Comment (LukasReschke) [inline ajax/event/edit.form.php]: Move to beginning of file.

File: ajax/event/edit.form.php [hunk 2]
```diff
@@ -1,18 +1,81 @@
...
+// no, we do not.
+} else {
+
+	// do we have a token?
+	if (!\OC::$session->exists('public_link_token'))
+		// nope, bail out!
+		OCP\User::checkLoggedIn();
+
+	// is the app enabled?
+	OCP\JSON::checkAppEnabled('calendar');
```
Comment (LukasReschke) [inline ajax/event/edit.form.php]: Delete

File: ajax/event/edit.form.php [hunk 3]
```diff
@@ -1,18 +1,81 @@
...
+	session_write_close();
+
+	// shareapi enabled?
+	if (\OC_Appconfig::getValue('core', 'shareapi_allow_links', 'yes') !== 'yes') {
+		header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline ajax/event/edit.form.php:27]: Use \OCP\Response

File: ajax/event/edit.form.php [hunk 4]
```diff
@@ -1,18 +1,81 @@
...
+
+	// get the data
+	$linkItem = OCP\Share::getShareByToken(
+		\OC::$session->get('public_link_token')
+	);
+
+	// did we get anything?
+	if (!is_array($linkItem) || !isset($linkItem['uid_owner'])) {
+		// nope! chuck testa!
+		header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline ajax/event/edit.form.php:39]: Use \OCP\Response

File: ajax/event/edit.form.php [hunk 5]
```diff
@@ -1,18 +1,81 @@
...
+		exit();
+	}
+
+	// resolve all the re-shares
+	$rootLinkItem = OCP\Share::resolveReShare($linkItem);
+	
+	// did we get anything?
+	if (!is_array($rootLinkItem) || !isset($rootLinkItem['uid_owner'])) {
+		// nnnneewp!
+		header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline ajax/event/edit.form.php:49]: Use \OCP\Response

File: ajax/event/edit.form.php [hunk 6]
```diff
@@ -1,18 +1,81 @@
...
+	}
+	
+	// do we have a password on this share?
+	if (isset($linkItem['share_with'])) {
+		// we're not going to check the password here, we're in AJAX mode
+		// what we can do is to check for 'public_link_authenticated' session var
+		if ( ! \OC::$session->exists('public_link_authenticated')
+				|| \OC::$session->get('public_link_authenticated') !== $linkItem['id']
+			) {
+				header('HTTP/1.0 401 Unauthorized');
```
Comment (LukasReschke) [inline ajax/event/edit.form.php:60]: Use \OCP\Response

File: ajax/event/edit.form.php [hunk 7]
```diff
@@ -1,18 +1,81 @@
...
+				exit();
+			}
+	}
 
-OCP\JSON::checkAppEnabled('calendar');
+	// just another check
+	if (!OC_Calendar_App::getCalendar($rootLinkItem['item_source'], true, true)) {
+		header('HTTP/1.0 403 Forbidden');
```
Comment (LukasReschke) [inline ajax/event/edit.form.php:67]: Use \OCP\Response

File: ajax/events.php [hunk 8]
```diff
@@ -1,30 +1,104 @@
...
-	}else{
-		if(OCP\Share::getItemSharedWithBySource('calendar', $id) === false){
-			OCP\JSON::encodedPrint(array());
-			exit;
+
+// is a user logged-in?
+if (OCP\User::isLoggedIn()) {
+
+	// is the app enabled?
+	OCP\JSON::checkAppEnabled('calendar');
```
Comment (LukasReschke) [inline ajax/events.php]: Move to top of file

File: ajax/events.php [hunk 9]
```diff
@@ -1,30 +1,104 @@
...
+	session_write_close();
+
+	// Look for the calendar id
+	$calendar_id = null;
+	if (strval(intval($_GET['calendar_id'])) == strval($_GET['calendar_id'])) { // integer for sure.
```
Comment (LukasReschke) [inline ajax/events.php]: Use ===

File: ajax/events.php [hunk 10]
```diff
@@ -1,30 +1,104 @@
...
+		if($calendarrow !== false) {
+			$calendar_id = $id;
+		}else{
+			if(OCP\Share::getItemSharedWithBySource('calendar', $id) === false){
+				OCP\JSON::encodedPrint(array());
+				exit;
+			}
 		}
 	}
+	$calendar_id = (is_null($calendar_id)?strip_tags($_GET['calendar_id']):$calendar_id);
```
Comment (LukasReschke) [inline ajax/events.php:28]: Test whether strip_tags is needed.

File: ajax/events.php [hunk 11]
```diff
@@ -1,30 +1,104 @@
...
+
+// no logged-in user? ookaay, do we have a token?
+} elseif (\OC::$session->exists('public_link_token')) {
+
+	// is the app enabled?
+	OCP\JSON::checkAppEnabled('calendar');
```
Comment (LukasReschke) [inline ajax/events.php]: Remove

File: ajax/events.php [hunk 12]
```diff
@@ -1,30 +1,104 @@
...
+	session_write_close();
+
+	// shareapi enabled?
+	if (\OC_Appconfig::getValue('core', 'shareapi_allow_links', 'yes') !== 'yes') {
+		header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline ajax/events.php:34]: Use \OCP\Response

File: ajax/events.php [hunk 13]
```diff
@@ -1,30 +1,104 @@
...
+		exit();
+	}
+	
+	// check if we're being asked for something we can provide
+	if ($_GET['calendar_id'] !== 'shared_events') {
+		header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline ajax/events.php:40]: Use \OCP\Response

File: ajax/events.php [hunk 14]
```diff
@@ -1,30 +1,104 @@
...
+		exit();
+	}
+
+	// resolve all the re-shares
+	$rootLinkItem = OCP\Share::resolveReShare($linkItem);
+	
+	// did we get anything?
+	if (!is_array($rootLinkItem) || !isset($rootLinkItem['uid_owner'])) {
+		// nope! chuck testa!
+		header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline ajax/events.php:62]: Use \OCP\Response

File: ajax/events.php [hunk 15]
```diff
@@ -1,30 +1,104 @@
...
+
+	// get the data
+	$linkItem = OCP\Share::getShareByToken(
+		\OC::$session->get('public_link_token')
+	);
+
+	// did we get anything?
+	if (!is_array($linkItem) || !isset($linkItem['uid_owner'])) {
+		// nope! chuck testa!
+		header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline ajax/events.php:52]: Use \OCP\Response

File: ajax/events.php [hunk 16]
```diff
@@ -1,30 +1,104 @@
...
+	}
+	
+	// do we have a password on this share?
+	if (isset($linkItem['share_with'])) {
+		// we're not going to check the password here, we're in AJAX mode
+		// what we can do is to check for 'public_link_authenticated' session var
+		if ( ! \OC::$session->exists('public_link_authenticated')
+				|| \OC::$session->get('public_link_authenticated') !== $linkItem['id']
+			) {
+				header('HTTP/1.0 401 Unauthorized');
```
Comment (LukasReschke) [inline ajax/events.php:73]: Use \OCP\Response

File: ajax/events.php [hunk 17]
```diff
@@ -1,30 +1,104 @@
...
+				exit();
+			}
+	}
+
+	// just another check
+	if (!OC_Calendar_App::getCalendar($rootLinkItem['item_source'], true, true)) {
+		header('HTTP/1.0 403 Forbidden');
```
Comment (LukasReschke) [inline ajax/events.php:80]: Use \OCP\Response

File: ajax/events.php [hunk 18]
```diff
@@ -1,30 +1,104 @@
...
+		exit();
+	}
+
+	// finally, get the calendar id
+	$calendar_id = $rootLinkItem['item_source'];
+	
+// no user, no token...
+} else {
+	header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline ajax/events.php:89]: Use \OCP\Response

File: share.php [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+ */
+ 
+OCP\App::checkAppEnabled('calendar');
+
+function calendar404($msg=null) {
+	$errorTemplate = new OCP\Template('calendar', 'part.404', '');
+	if ($msg !== null) $errorTemplate->assign('message', $msg);
+	$errorContent = $errorTemplate->fetchPage();
+
+	header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline share.php:18]: Use OCP\Response

File: share.php [hunk 20]
```diff
@@ -0,0 +1,191 @@
...
+ * This file is licensed under the Affero General Public License version 3 or
+ * later.
+ * See the COPYING-README file.
+ */
+ 
+OCP\App::checkAppEnabled('calendar');
+
+function calendar404($msg=null) {
+	$errorTemplate = new OCP\Template('calendar', 'part.404', '');
+	if ($msg !== null) $errorTemplate->assign('message', $msg);
```
Comment (LukasReschke) [inline share.php]: Use curly braces

File: share.php [hunk 21]
```diff
@@ -0,0 +1,191 @@
...
+
+	header('HTTP/1.0 404 Not Found');
+	$tmpl = new OCP\Template('', '404', 'guest');
+	$tmpl->assign('content', $errorContent);
+	$tmpl->printPage();
+	exit();
+}
+
+function calendar403() {
+	header('HTTP/1.0 403 Forbidden');
```
Comment (LukasReschke) [inline share.php:26]: Use OCP\Response

File: share.php [hunk 22]
```diff
@@ -0,0 +1,191 @@
...
+	$tmpl = new OCP\Template('', '403', 'guest');
+	$tmpl->printPage();
+	exit();
+}
+
+if (\OC_Appconfig::getValue('core', 'shareapi_allow_links', 'yes') !== 'yes')
```
Comment (LukasReschke) [inline share.php]: Use curly braces

File: share.php [hunk 23]
```diff
@@ -0,0 +1,191 @@
...
+			$type = OC_Calendar_Export::CALENDAR;
+		// event
+		} else {
+			OCP\Util::writeLog('calendar', __FILE__ . ' : ' . __METHOD__, OCP\Util::ERROR);
+			$data = OC_Calendar_App::getEventObject($rootLinkItem['item_source'], true, true);
+			$type = OC_Calendar_Export::EVENT;
+		}
+		if(!$data) {
+			OCP\Util::writeLog('share', 'forbidden!', \OCP\Util::ERROR);
+			header('HTTP/1.0 403 Forbidden');
```
Comment (LukasReschke) [inline share.php:129]: Use OCP\Response

File: share.php [hunk 24]
```diff
@@ -0,0 +1,191 @@
...
+			exit;
+		}
+		header('Content-Type: text/calendar');
```
Comment (LukasReschke) [inline share.php:132]: Use OCP\Response

File: share.php [hunk 25]
```diff
@@ -0,0 +1,191 @@
...
+		header('Content-Disposition: inline; filename=' . str_replace(' ', '-', $data['displayname']) . '.ics');
```
Comment (LukasReschke) [inline share.php:133]: Use OCP\Response

File: share.php [hunk 26]
```diff
@@ -0,0 +1,191 @@
...
+		// export the data
+		// if it is a link-shared concrete event, ignore security
+		// calendars should be shared *with* security enabled, so as to not divulge private/busy events
+		echo OC_Calendar_Export::export($rootLinkItem['item_source'], $type, ($type !== OC_Calendar_Export::EVENT) );
```
Comment (LukasReschke) [inline share.php]: Using echo is ugly

File: share.php [hunk 27]
```diff
@@ -0,0 +1,191 @@
...
+		OCP\Util::addStyle('3rdparty/timepicker', 'jquery.ui.timepicker');
+		OCP\Util::addscript('calendar', 'calendar');
+		OCP\Util::addStyle('calendar', 'style');
+		OCP\Util::addStyle('calendar', 'tooltips');
+		OCP\Util::addscript('', 'jquery.multiselect');
+		OCP\Util::addStyle('', 'jquery.multiselect');
+		OCP\Util::addscript('calendar','jquery.multi-autocomplete');
+		OCP\Util::addscript('','tags');
+		OCP\Util::addscript('calendar','on-event');
+		OCP\Util::addscript('calendar','settings');
```
Comment (LukasReschke) [inline share.php:154]: This bloats the code - can't it be added to the template?

File: templates/event.php [hunk 28]
```diff
@@ -0,0 +1,309 @@
...
+ * later.
+ * See the COPYING-README file.
+ */
+
+// this also checks if we're properly called from share.php
+if (!function_exists('calendar404')) {
+	$errorTemplate = new OCP\Template('calendar', 'part.404', '');
+	$errorContent = $errorTemplate->fetchPage();
+
+	header('HTTP/1.0 404 Not Found');
```
Comment (LukasReschke) [inline templates/event.php:15]: Use \OCP\Response

File: templates/event.php [hunk 29]
```diff
@@ -0,0 +1,309 @@
...
+	$tmpl->assign('content', $errorContent);
+	$tmpl->printPage();
+	exit();
+}
+
+$id = $_['link_shared_event']['item_source'];
+$data = OC_Calendar_App::getEventObject($id, false, false);
+
+// whoops
+if(!$data) calendar404();
```
Comment (LukasReschke) [inline templates/event.php]: Use curly braces

File: templates/part.internalshare.php [hunk 30]
```diff
@@ -0,0 +1,102 @@
...
+				<!--
+					"can edit" checkbox
+						- if the item type allows for precise sharing settings (update, create, delete), setting/un-setting all three at the same time via JS (and reflecting their state)
+						- if the item type allows only for editable/uneditable (actually, just update) setting, the "update" checkbox
+					this is checked via $_['basic_edit_options'] flag (if set and true, only editable/uneditable setting available)
+
+					this checkbox is available only when JS is enabled
+					otherwise the noscript tag makes space for a CSS/HTML "checkbox" showing the state of the create/update/delete checkboxes
+				-->
+				<noscript class="share-can-edit-space"><!--</noscript>
```
Comment (LukasReschke) [inline templates/part.internalshare.php:55]: noscript? we do not support users without JS.

File: templates/part.showevent.php [hunk 31]
```diff
@@ -1,15 +1,50 @@
...
+<?php if (isset($_['link_shared_event'])) { ?>
+<header id="linksharedinfo">
+		<div class="header-right">
+			<span id="details"><?php p($l->t('shared by %s', array($_['link_shared_event']['uid_owner']))) ?></span>
+		</div>
+		<a href="<?php print_unescaped(link_to('', 'index.php')); ?>" title="" id="owncloud"><img class="svg"
+			src="<?php print_unescaped(image_path('', 'logo-wide.svg')); ?>" alt="<?php p($theme->getName()); ?>"
+		/></a>
+		<div id="logo-claim" style="display:none;"><?php p($theme->getLogoClaim()); ?></div>
+		<div><?php p($l->t('Event')) ?> &quot;<?php p($_['link_shared_event']['item_target'])?>&quot;; <?php p($l->t('download or use in your calendar application:'))?> <a class="download-link" href="<?php echo $_['link_shared_event_url']; ?>&amp;download"><?php p($l->t('Download'))?></a></div>
```
Comment (LukasReschke) [inline templates/part.showevent.php]: Do not use echo

File: templates/part.showevent.php [hunk 32]
```diff
@@ -1,15 +1,50 @@
...
+			$ex[1] = $ex[0];
+			$ex[0] = "Other";
+		}
+		if ($continent!=$ex[0]):
+			if ($continent!="") print_unescaped('</optgroup>');
+			print_unescaped('<optgroup label="'.OC_Util::sanitizeHTML($ex[0]).'">');
+		endif;
+		$city=strtr($ex[1], '_', ' ');
+		$continent=$ex[0];
+		print_unescaped('<option value="'.OC_Util::sanitizeHTML($timezone).'"'.($_['timezone'] == $timezone?' selected="selected"':'').'>'.OC_Util::sanitizeHTML($city).'</option>');
```
Comment (LukasReschke) [inline templates/part.showevent.php]: Use the public interface

**Comment by jbtbnl** [comment]
@LukasReschke this branch is in owncloud/calendar so it would be cool if you could commit a patch for the issues you noticed!

File: templates/part.internalshare.php (hunk 30 above)
Comment (rysiekpl) [inline templates/part.internalshare.php:55]: That's future-proofing. JS, especially in a way it's being used in ownCloud, is problematic to visually-impaired users. This is a big problem -- I know of at least 2 possible large ownCloud deployments that in the end went with Google Apps instead just due to this.

Comment (jancborchardt) [inline templates/part.internalshare.php:55]: @rysiekpl can you put me in contact with those organizations? We want to improve accessibility, and especially when it’s a matter of improving the HTML, I want to do that.

**Comment by MorrisJobke** [comment]
@jbtbnl Thanks for your effort! I have to travel the whole day and therefore am unable to review this in a proper way.

File: templates/part.internalshare.php (hunk 30 above)
Comment (rysiekpl) [inline templates/part.internalshare.php:55]: These are Polish organisations, and the person best fit to speak for both them on this matter is not really fluent in English. I will be happy to work as a liaison, though. If you send me an e-mail (rysiek at fwioo.pl), we can get in touch.

**Comment by karlitschek** [comment]
ping

File: ajax/event/edit.form.php (hunk 3 above)
Comment (georgehrke) [inline ajax/event/edit.form.php:27]: \OCP\Response does not contain a setResponseCode or whatsoever method

**Comment by georgehrke** [comment]
@jbtbnl We got a deadline extension 'til friday for this certain feature. 
Can you take care of the design issues I mentioned?
I'll take care of the issues mentioned by @LukasReschke

**Comment by jbtbnl** [comment]
@georgehrke OK, that should work!

File: ajax/event/edit.form.php (hunk 3 above)
Comment (LukasReschke) [inline ajax/event/edit.form.php:27]: Ewww, why is https://github.com/owncloud/core/blob/master/lib/private/response.php#L56 not public? :-/

Might be a good idea to migrate it. It's used by tons of apps...

**Comment by jbtbnl** [comment]
@georgehrke @DeepDiver1975 

The current calendar/master uses a component from Core to present the internalShare dialog but as it seems @rysiekpl didn't extend or reuse it for the link sharing.

The UI part for link sharing seems to be already available in Core because Files also has link sharing. But I cannot find the code that is providing it or how to use it.

File: ajax/event/edit.form.php (hunk 3 above)
Comment (MorrisJobke) [inline ajax/event/edit.form.php:27]: I guess it lacks testing due to appframework integration/migration https://github.com/owncloud/core/blob/master/lib/public/appframework/http/response.php#L143

**Comment by libasys** [comment]
With this PR you can use the shareApi for "Shared by Link" of the core:
https://github.com/owncloud/core/pull/7731
Have a look into the master!
And this 2 options to check the rights:

```
\OC_Appconfig::getValue('core', 'shareapi_allow_mail_notification', 'yes');
\OC_Appconfig::getValue('core', 'shareapi_allow_links', 'yes');
```

then use a hidden input field to check  in the calendar form e.g.:

```
<input type="hidden" name="mailNotificationEnabled" id="mailNotificationEnabled" value="<?php p($_['mailNotificationEnabled']) ?>" />
<input type="hidden" name="allowShareWithLink" id="allowShareWithLink" value="<?php p($_['allowShareWithLink']) ?>" />
```

**Comment by jbtbnl** [comment]
@libasys thanks for informing! I'll take a look at it.

File: templates/part.showevent.php (hunk 31 above)
Comment (georgehrke) [inline templates/part.showevent.php]: is it safe to use print_unescaped here?

**Comment by karlitschek** [comment]
so what is the status here?

**Comment by jbtbnl** [comment]
@karlitschek the client side is way more broken than I expected. I would not recommend to merge the  current implementation. The share dialog as seen in core should be reused but I don't know where to start with that and don't have a lot of time available...

**Comment by libasys** [comment]
https://github.com/owncloud/core/pull/9592

with this pr we only have to look for the public side and code them!
Working examples:

Share event via link:
https://ssl.webpack.de/demo70.libasyscloud.de/owncloud/public.php?service=event&t=3f5bb544548542b821835848b6426072

Share calendar via link:
https://ssl.webpack.de/demo70.libasyscloud.de/owncloud/public.php?service=calendar&t=41aac5cb9ed3d249833fde315281de87

**Comment by tele81** [comment]
is it ready for oc 7.0.1?

best regards

**Comment by jancborchardt** [comment]
@georgehrke please comment …

**Comment by georgehrke** [comment]
from the server side yes,
i am not sure about the ui though. ping @jbtbnl

**Comment by DeepDiver1975** [comment]
- [ ] rebase
- [ ] squash
- [ ] have a look at scrutinizer issues

**Comment by jbtbnl** [comment]
@jancborchardt @georgehrke the UI is broken, unfortunately. See my comment in https://github.com/owncloud/calendar/pull/440#issuecomment-46015942

**Comment by CorentinS6** [comment]
Hello,
Is this request still in dev ? I'm intersted in.
Regards.

**Comment by georgehrke** [comment]
@jbtbnl How do we want to proceed here?
Try to fix the UI or maybe wait for https://github.com/owncloud/core/pull/9592 to be merged?

**Comment by crti** [comment]
In the oC-Forum a user published a workaround script to get the basic feature with CURL: http://forum.owncloud.org/viewtopic.php?f=29&t=24234 
I'm thankful to every experienced oC-coder investing in public calendar sharing for the core and GUI. This feature is the one I mostly desire.

**Comment by farson2003** [comment]
Any idea when it could be implemented to share of the calendar webinterface via link?

**Comment by rysiekpl** [comment]
@farson2003 apparently never; or rather: it _has been_ implemented for months now, just not merged into mainline ownCloud calendar.

ownCloud developers do not seem interested in merging this, and I have no more time to lose on work that is not being merged. This [drags on for months now](https://github.com/owncloud/calendar/pull/308), with no clear end in sight.

**Comment by georgehrke** [comment]
@rysiekpl We are as eager to merge this as you, but the interface broke when we moved the calendar-list to the sidebar. @jbtbnl and me are not working full-time on the calendar. Everything we do in the calendar we do in our leisure time. It would be very much appreciated if you could help looking into fixing this. 

@jbtbnl please comment on https://github.com/owncloud/calendar/pull/440#issuecomment-54801355

**Comment by jbtbnl** [comment]
@georgehrke I think it's the most feasible to split off the back-end of this PR. Then rebase and merge the back-end into the rework branch, and re-develop the front-end part.

That will ensure that the back-end code can be properly reviewed before merging (as far as I can see there has not been a full review on that yet) and it will also ensure that the amazing job that @rysiekpl has done is not wasted.

**Comment by farson2003** [comment]
@DeepDiver1975 @georgehrke @karlitschek @jbtbnl  @MorrisJobke  @jancborchardt 

@rysiekpl wrote "...apparently never; or rather: it has been implemented for months now, just not merged into mainline ownCloud calendar.".

There are so many scenarios for this implementation. I.e. many organizations would love to share a public calendar via a pubilc OC-calendar webinterface.

This public calendar could even be implemented in a frame on a company's or organization's website... 
...and so on. ... many scenarios.

It would be a pitty if ths feature wasn't implemented soon.

**Comment by farson2003** [comment]
Is there any positive update on this?

**Comment by farson2003** [comment]
@DeepDiver1975 @georgehrke @karlitschek @jbtbnl @MorrisJobke @jancborchardt

@rysiekpl 
sorry, but is there an approx. timeframe when this will be tackled?

**Comment by georgehrke** [comment]
We will try out best to get it in with oC 8.

Please excuse my brevity and typos.

**Comment by rysiekpl** [comment]
@farson2003 I'm sorry, but given the time investment I did on this and the most possible outcome (most of my work being scrapped due to interface changes, and the sharing interface being reverted to JS-based one that is inaccessible to the visually-impaired), I have no interest in pursuing this pull request. Simply put, my time can be much better spent on pretty much anything else.

**Comment by thepfy** [comment]
Thank you @rysiekpl for your efforts, I wish I could code better I would give it a go if I could.

but its a real shame! I think a number of owncloud users really need this. I will have to find another product.

**Comment by mdimjasevic** [comment]
@georgehrke so, hopefully the public calendar sharing feature will appear in ownCloud 8 according to the milestones. When is that milestone planned to happen?

This is a so-long awaited feature by so many ownCloud users... it's been almost a year since @rysiekpl has implemented it.

**Comment by joeplus** [comment]
+1 @mdimjasevic

**Comment by ghost** [comment]
Hi,

@mdimjasevic 

> so, hopefully the public calendar sharing feature will appear in ownCloud 8 according to the milestones. When is that milestone planned to happen?

This pull request isn't merged yet and OC8 is already in feature freeze: http://owncloud.10557.n7.nabble.com/feature-freeze-td13735.html so it is probably very unlikely that this will end up in OC8.

**Comment by MorrisJobke** [comment]
@georgehrke Can you please have a look and fix and merge this. It's just annoying that so much effort is been added to this and nothing happens.

**Comment by georgehrke** [comment]
> @georgehrke Can you please have a look and fix and merge this. It's just annoying that so much effort is been added to this and nothing happens.

I'm also not happy that this hasn't been merged yet and I am working on it, but as @jbtbnl stated: the interface is completely broken and that's nothing you can fix that easily.

**Comment by jbtbnl** [comment]
@georgehrke @MorrisJobke that's why I [proposed](https://github.com/owncloud/calendar/pull/440#issuecomment-60053355) to split off the backend part of this PR and review/merge that.

In the current state it is counterproductive to want to merge both the back- and frontend at once.

**Comment by MorrisJobke** [comment]
@jbtbnl @georgehrke Okay. Then split it up. But wasting work that is done for free isn't motivating for collaborators ;) I would love to see this in a future release :)

**Comment by farson2003** [comment]
You are joking me...still not coming in OC8?
Sad enough.

**Comment by flexibeast** [comment]
i'm very eager for the sort of functionality mentioned in #245, so was wondering if someone could provide an update re. the status of this PR and the amount of work required for it to be merged?

**Comment by jospoortvliet** [comment]
Yeah, after a year after this PR got posted, it'd be nice to get it in. Seeing the long conversations I guess it should have been split up in the beginning and merged in pieces back then - or just merged as it was and stuff fixed up after that... This is demotivating and depressing to read.

**Comment by MorrisJobke** [comment]
@PVince81 @jancborchardt Are you interested in a review session next week in Nuremberg? I want this is. It's just stupid to always complain about. We should simply do it. I will write it onto our list.

**Comment by MorrisJobke** [comment]
It's already in. Lets do this. I'm looking forward to this :)

**Comment by farson2003** [comment]
@MorrisJobke  @PVince81 @jancborchardt 
Has anything happened in Nurenberg with a promising outcome regarding this feature?
I assume many "followers" can't wait...
Thank you

**Comment by MorrisJobke** [comment]
@farson2003 Sadly not :(

**Comment by MorrisJobke** [comment]
@PVince81 Is calendar working now with the updated Sabre?

**Comment by georgehrke** [comment]
@MorrisJobke yep, we merged the fixes ~2 weeks ago

**Comment by crti** [comment]
May I ask as I are not able to sort or understand the comments from @MorrisJobke and @georgehrke: what is the status or in what public/stable version of OwnCloud it is expected (or already there?) to have public calendar sharing available in the GUI for usage?

**Comment by farson2003** [comment]
@crti  @MorrisJobke @PVince81 @jancborchardt  @georgehrke
I agree with @crti in that I do not understand the comments reegarding the availability of this puclic calendar sharing. 
What is the status of this feature?

**Comment by phsc84** [comment]
Another calendar app is currently in work, that will provide public calendar sharing:
https://forum.owncloud.org/viewtopic.php?t=28081

Who knows, maybe it will get the new default calendar app for owncloud, if it is better than this one.

**Comment by libasys** [comment]
@phsc here you can download the calendar app for version owncloud 8.1:
https://github.com/libasys/ownCloud-8.1-Calendar-App
Betatester welcome, pleas report bugs for this calendar app on:
https://github.com/libasys/ownCloud-8.1-Calendar-App/issues

Thanks!

**Comment by DeepDiver1975** [comment]
I'm closing this in favor of #952

**Comment by jancborchardt** [comment]
Everyone here, please review and test https://github.com/owncloud/calendar/pull/952

**Pull request closed without merging**